### PR TITLE
Listen for turbolinks:load event

### PR DESCRIPTION
**Pull request by twalpole**
This adds the turbolinks:load event that Turbolinks5 uses rather than page:change

### CODE REVIEW

**Comment by javan** [comment]
:+1: to adding `turbolinks:load`. Mind looking into the test failures?

**Comment by twalpole** [comment]
Looking into it now

**Comment by twalpole** [comment]
@javan it appears to be some sort of conflict with the test for the old turbolinks page:change event since either one works separately but if they're both run the last one fails.  I'm not familiar with blade and don't know what about the environment isn't getting reset

**Comment by twalpole** [comment]
@javan Ok - I went ahead and combined the two turbolinks tests into one since theres no way I can figure out to remove the event listeners added in the previous test (that is obvious anyway).  I also don't think the original test was testing what it was supposed to be since the triggering of DOMContentLoaded after the page is already loaded, immediately calls the callback - so whether or not the turbolinks events were being listened for was irrelevant.  I've changed the test to add new elements before triggering the turbolinks events so it's only the turbolinks events updating those elements (and possibly existing time:elapse events of course)

File: app/assets/javascripts/local_time.js.coffee [hunk 1]
```diff
@@ -200,7 +200,8 @@ update = (callback) ->
   document.addEventListener "time:elapse", callback
 
   if Turbolinks?.supported
-    document.addEventListener "page:update", callback
+    for event in ["page:update", "turbolinks:load"]
+      document.addEventListener event, callback
```
Comment (javan) [inline app/assets/javascripts/local_time.js.coffee]: Not exactly related to this change, but simply checking `Turbolinks.supported` will miss the jQuery `ajaxSuccess` event when using TL5. TL classic [triggers `page:update` on `ajaxSuccess`](https://github.com/turbolinks/turbolinks-classic/blob/master/lib/assets/javascripts/turbolinks.coffee#L654-L656) and TL5 does not (unless you include the [compatibility shim](https://github.com/turbolinks/turbolinks/blob/master/src/turbolinks/compatibility.coffee)).

So the ideal condition, in somewhat pseudo code, would be:
```coffee
if Turbolinks?.supported
  if Turbolinks.version < 5 or Turbolinks.version >= 5 and Turbolinks.compatibilityMode
    # Add page:update listener
  else
    # Add turbolinks:load listener
    # Add jQuery ajaxSuccess listener
else
  # Add jQuery ajaxSuccess listener
```
/cc @sstephenson @packagethief

Comment (sstephenson) [inline app/assets/javascripts/local_time.js.coffee]: IMO we should remove the Turbolinks hooks entirely and provide a MutationObserver that automatically processes `<time>` elements.

**Comment by javan** [comment]
Added a `MutationObserver` to #55, which I'm planning to release as version 2.0. Give a try if you'd like.

**Comment by meowsus** [comment]
We're having issues in our environment that this PR should fix, since Turbolinks changed it's events and locale_time has yet to start listening to the new events. 

Is this PR in a mergable state?

**Pull request closed without merging**